src/store/api/worldUsers.ts

```diff
@@ -0,0 +1,193 @@
+  //   place/working, we want to refactor it to split the data in a 'smarter'/more efficient way.
+  worldUsers: WithId<User>[];
+  // @debt worldUsersById should be Partial<Record<string, WithId<User>>>, current kept as is for legacy reasons, but it's technically a typing bug
+  worldUsersById: Record<string, WithId<User>>;
+  // @debt worldUserLocationsById should be Partial<Record<string, WithId<UserLocation>>>, current kept as is for legacy reasons, but it's technically a typing bug
+  worldUserLocationsById: Record<string, WithId<UserLocation>>;
+}
+
+export const initialState: Readonly<WorldUsersData> = {
+  worldUsers: [],
+  worldUsersById: {},
+  worldUserLocationsById: {},
+};
+
+// TODO: https://redux-toolkit.js.org/rtk-query/api/createApi
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#streaming-data-with-no-initial-request
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#adding-meta-information-to-queries
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#constructing-a-dynamic-base-url-using-redux-state
+//   Can we use this to read the sovereignVenueId (or even currentVenueId or maybe all relatedVenueIds as needed) directly out of the state?
+//   And even if we can, do we want to? Or keep passing it in through WorldUsersApiArgs? (maybe the latter, as I think then it
+//     is used for/relates to the cache key for this data)
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/migrating-to-rtk-query
+export const worldUsersApi = createApi({
+  reducerPath: "worldUsersApi",
+  baseQuery: fakeBaseQuery(),
+  endpoints: (builder) => ({
+    worldUsers: builder.query<WorldUsersData, WorldUsersApiArgs>({
+      // The query is not relevant here as the data will be provided via streaming updates.
+      // A queryFn returning an empty array is used, with contents being populated via
+      // streaming updates below as they are received.
+      queryFn: () => ({
+        data: initialState,
+        // TODO: we can also return error and/or meta here as well as data. Do we need to be able to?
+        // error: undefined,
+        // meta: undefined,
+      }),
+      async onCacheEntryAdded(
+        { relatedLocationIds },
+        { updateCachedData, cacheEntryRemoved }
+      ) {
+        // TODO: use api/* layer for this?
+        const usersCollectionRef = firebase.firestore().collection("users");
+
+        const worldUsersQuery = usersCollectionRef.where(
+          "enteredVenueIds",
+          "array-contains-any",
+          relatedLocationIds
+        );
+
+        const unsubscribeListener = worldUsersQuery.onSnapshot((snapshot) => {
+          // TODO: implement this properly + cleanup this placeholder stuff
+          const added = snapshot
+            .docChanges()
+            .filter((docChange) => docChange.type === "added").length;
```

Comment:
do we want to introduce a helper function for `added`, `removed` and `modified` ? code seems to repeat there
also, maybe move 'added', 'removed' etc strings to consts somewhere?

Reply:
Are you talking about in these `.filter` functions? Or..? If so, they are placeholder to be removed before this is merged properly. If the latter, can you please expand on your thoughts for what the helper function would entail.

Reply:
yep, talking about .filter ones and moving shared logic `snapshot.docChanges().filter` into some func that would be reused for these `added`, `removed` etc consts

Reply:
So these specific filters will be removed. They're just debug code. Not sure that we gain anything by making the `"added"`/etc consts, TypeScript already restricts them to only valid values, and these come from the library itself, so no reason for us to make a new enum/etc.

If the lib exports an enum I could potentially use that here, but I think it's probably fine as is.